### ISSUE TITLE
Ascend device training questions about the accuracy of training

### ISSUE THREAD

**Issue by 1737686924** [ultralytics/ultralytics#16829]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I simply made some training modifications and trained on the Ascend device, and found that the training accuracy of object detection was normal.
This is an example of image classification training
![image](https://github.com/user-attachments/assets/1b8122fb-4c13-489e-a962-fd33993f9de9)
，Can you give us some help，We want to support more devices

**Comment by UltralyticsAssistant** [comment]
👋 Hello @1737686924, thank you for reaching out and for your interest in Ultralytics 🚀!

If you're having issues with your custom training on Ascend devices, please ensure you're following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/). Providing detailed information such as dataset samples, training logs, and any custom modifications can help us assist you better.

For real-time support and to discuss with other users, feel free to join our [Discord](https://ultralytics.com/discord) community 🎧. Alternatively, if you prefer more in-depth conversations, [Discourse](https://community.ultralytics.com) is the place to go. You can also connect with fellow users on our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

Make sure to upgrade to the latest `ultralytics` package to ensure you're on the newest version which might have important fixes:

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in up-to-date verified environments. You might find using these environments useful for debugging or enhancing performance:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultraviolytics.com) for more information on Python and CLI usage examples, which might already have answers to some of your questions.

We encourage you to provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us debug any specific issues you encounter during training.

An Ultralytics engineer will also assist you soon.

**Comment by Y-T-G** [comment]
Did the AMP checks in the beginning pass or fail?

**Comment by 1737686924** [comment]
> 开始时的 AMP 检查是通过还是失败？

With AMP, the check passed, but it encountered a gradient explosion，However, the training accuracy has not improved

**Comment by Y-T-G** [comment]
Can you try using `amp=False` and `half=False`?

**Comment by 1737686924** [comment]
> Can you try using `amp=False` and `half=False`?


![image](https://github.com/user-attachments/assets/2ca3f1e1-dbd8-42db-85ad-67a7d2356348)
I have tried it, and I found that there is a problem with the loss, and the accuracy has increased strangely, and it suddenly rises

**Comment by glenn-jocher** [comment]
It sounds like there might be an issue with your learning rate or data. Try adjusting the learning rate or reviewing your dataset for inconsistencies. If the problem persists, consider checking your model's architecture or hyperparameters.

**Comment by Y-T-G** [comment]
You can try opening an issue on Ascend PyTorch repo since it's probably an issue with their implementation

**Comment by lovedream-ms** [comment]
抱歉我有一个问题，Ascend 设备训练也有GPU嘛

**Comment by glenn-jocher** [comment]
No, Ascend devices use NPUs (Neural Processing Units) rather than GPUs. They're Huawei's AI acceleration hardware specifically designed for deep learning workloads, but with a different architecture than NVIDIA GPUs. The training issues being discussed are likely related to this architectural difference, which may affect how PyTorch operations are implemented on Ascend hardware.

**Comment by glenn-jocher** [comment]
Hi @1737686924 and @lovedream-ms, a related Huawei Ascend update: native Ascend export, inference, and validation support has now shipped in Ultralytics. Models can be exported directly to `.om` format with `format=ascend`, and [Ultralytics Platform](https://platform.ultralytics.com) now provides managed exports for Ascend310P1, Ascend310P3, Ascend310B1, and Ascend310B4.

The complete workflow is documented in the [Huawei Ascend integration guide](https://docs.ultralytics.com/integrations/ascend/).

This update does not change Ascend training behavior or resolve the training-accuracy problem discussed in this issue; it applies specifically to export, inference, and validation.

![Ultralytics Platform Huawei Ascend export](https://gist.githubusercontent.com/glenn-jocher/ddb7e1146039d729ddb891c6bc261e6b/raw/0af8f891222ccd17f783f8185b7c36f690e5b87d/ultralytics-platform-huawei-ascend-export.png)
